lib/pure/httpclient.nim

```diff
@@ -297,44 +317,48 @@ proc newProxy*(url: string, auth = ""): Proxy =
 
 proc newMultipartData*: MultipartData =
   ## Constructs a new ``MultipartData`` object.
-  MultipartData(content: @[])
-
+  MultipartData()
 
 proc `$`*(data: MultipartData): string {.since: (1, 1).} =
   ## convert MultipartData to string so it's human readable when echo
   ## see https://github.com/nim-lang/Nim/issues/11863
-  const prefixLen = "Content-Disposition: form-data; ".len
-  for pos, item in data.content:
-    result &= "------------------------------  "
-    result.addInt pos
-    result &= "  ------------------------------\n"
-    result &= item[prefixLen .. item.high]
-
-proc add*(p: var MultipartData, name, content: string, filename: string = "",
-          contentType: string = "") =
-  ## Add a value to the multipart data. Raises a `ValueError` exception if
-  ## `name`, `filename` or `contentType` contain newline characters.
-
+  const sep = "-".repeat(30)
+  for pos, entry in data.content:
+    result.add(sep & center($pos, 3) & sep)
+    result.add("\nname=\"" & entry.name & "\"")
+    if entry.isFile:
+      result.add("; filename=\"" & entry.filename & "\"\n")
+      result.add("Content-Type: " & entry.contentType)
+    result.add("\n\n" & entry.content & "\n")
+
+proc add*(p: MultipartData, name, content: string, filename: string = "",
+          contentType: string = "", stream = true) =
+  ## Add a value to the multipart data. Raises a ``ValueError`` exception if
```

Comment:
```suggestion
  ## Add a value to the multipart data.
  ##
  ## Raises a ``ValueError`` exception if
```

Reply:
This is a nit, but in general, please try to separate doc comments into (the empty `##` line is important):

```
## Short summary of what proc does
##
## More detailed information about what proc does.
##
## Details about exceptions
##
## Other misc info.
```